app/src/main/java/com/naman14/timberx/notifications/NotificationsModule.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2019 Naman Dwivedi.
+ *
+ * Licensed under the GNU General Public License v3
+ *
+ * This is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ */
+package com.naman14.timberx.notifications
+
+import android.app.Application
+import android.app.NotificationManager
+import android.content.Context
+import com.naman14.timberx.util.extensions.systemService
+import org.koin.dsl.module.module
+
+val notificationModule = module {
```

Comment:
Here's the magic of KOIN - you organize components into modules, which provide a set of instances.

`factory` means it constructs a new instance every time it's requested lazily; `single` means the lazily created instance will survive through the whole life of the application.

You can also use custom scopes and do some other fancy stuff but that's not important here.